cloudigrade/api/models.py

```diff
@@ -19,6 +21,83 @@
 logger = logging.getLogger(__name__)
 
 
+class UserManager(BaseUserManager):
+    """Manager class for the Api User model."""
+
+    def create_user(
+        self,
+        account_number=None,
+        org_id=None,
+        password=None,
+        date_joined=None,
+        is_active=True,
+        is_superuser=False,
+    ):
+        """create_user creates a user, requires at least an account_number or org_id."""
+        if not account_number and not org_id:
+            raise ValueError("Users must have at least an account_number or org_id")
+
+        user = self.model(account_number=account_number, org_id=org_id)
+        if password:
+            user.set_password(password)
+        else:
+            user.set_unusable_password()
+
+        if date_joined:
+            user.date_joined = date_joined
+
+        user.is_active = is_active
+        user.is_superuser = is_superuser
+        user.save(using=self._db)
+        return user
+
+
+class User(AbstractBaseUser):
+    """Class for our Api User model."""
+
+    def user_uuid():
+        """Return a UUID string to use for newly created Api users."""
+        return str(uuid.uuid4())
+
+    uuid = models.TextField(
+        verbose_name="User Unique Id",
+        max_length=150,
+        unique=True,
+        blank=False,
+        default=user_uuid,
+    )
+    account_number = models.TextField(
+        verbose_name="Account Number",
+        max_length=150,
+        unique=True,
+        null=True,
```

Comment:
@brad I just remembered I should probably add db_index=True to both account_number and org_id.

Reply:
Agreed, good idea!